src/data/mv_polynomial/equiv.lean

```diff
@@ -319,6 +321,257 @@ begin
+    by_cases c_mj' : m j' = 0,
+    { simp only [r, finsupp.cons_succ, c_mj', if_false, eq_self_iff_true, not_true] },
+    { simp only [r, finsupp.cons_succ, c_mj', if_true, not_false_iff],
+      congr,
+      ext,
+      by_cases c_a : a = 0,
+      { simp [c_a, finsupp.cons_zero, finsupp.single, c_j] },
+      { simp only [finsupp.coe_tsub, pi.sub_apply],
+        rw ←fin.succ_pred a c_a,
+        repeat {rw finsupp.cons_succ},
+        simp [finsupp.single, c_j],
+        congr } } },
+end
+
+/-- This gives the relation between the coefficients of the coefficients of `fin_succ_equiv R n f`
+    and the coefficients of `f`. -/
+lemma fin_succ_equiv_coeff_coeff {n : ℕ} {R : Type u} [comm_semiring R]
+  (m : fin n →₀  ℕ) (f : mv_polynomial (fin (n + 1)) R ) (i : ℕ) :
+  coeff m (polynomial.coeff (fin_succ_equiv R n f) i) = coeff (finsupp.cons i m) f :=
+begin
+  revert i m,
+  apply induction_on f,
+  apply fin_succ_equiv_coeff_coeff_C,
+  intros p q hp hq i m,
+  simp only [(fin_succ_equiv R n).map_add, polynomial.coeff_add, coeff_add, hp, hq],
+  apply fin_succ_equiv_coeff_coeff_case_p_X,
+end
+
+lemma eval_eq_eval_mv_eval' {n : ℕ} {R : Type u} [comm_semiring R] (s : fin n → R) (y : R)
+  (f : mv_polynomial (fin (n + 1)) R) : eval (fin.cons y s : fin (n + 1) → R) f
+  = polynomial.eval y (polynomial.map (eval s) ((fin_succ_equiv R n) f)) :=
+begin
+  apply induction_on f,
+  intro a,
+  simp,
+  intros p q hp hq,
+  simp only [(fin_succ_equiv R n).map_add, ring_hom.map_add, polynomial.map_add,
+             polynomial.eval_add],
+  congr,
+  exact hp,
+  exact hq,
+  intros p j h,
+  simp only [(fin_succ_equiv R n).map_mul, eval_X, polynomial.map_mul, ring_hom.map_mul,
+             polynomial.eval_mul],
+  congr,
+  exact h,
+  clear h f,
+  simp only [fin_succ_equiv_apply, eval₂_hom_X'],
+  by_cases c : j = 0,
+  { rw c,
+    simp [fin.cons_zero] },
+  have c' : j ≠ 0 := by simpa only [ne.def],
+  rw [←fin.succ_pred j c', fin.cons_succ],
+  simp,
+end
+
+lemma coeff_eval_eq_eval_coeff {n : ℕ} {R : Type u} [comm_semiring R] (s' : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) (i : ℕ) :
+  polynomial.coeff (polynomial.map (eval s') f) i =  eval s' (polynomial.coeff f i) :=
+by simp only [polynomial.coeff_map]
+
+/- TODO: what is a good name for this one? -/
+lemma support_eval {n : ℕ} {R : Type u} [comm_semiring R] (s' : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) (i : ℕ) :
+  (polynomial.map (eval s') f).support ⊆ f.support :=
+begin
+  intros i hi,
+  simp only [polynomial.mem_support_iff, polynomial.coeff_map, ne.def] at hi,
+  by_contradiction c,
+  simp only [polynomial.mem_support_iff, not_not, ne.def] at c,
+  rw c at hi,
+  simpa using hi,
+end
+
+lemma degree_eval_le_degree {n : ℕ} {R : Type u} [comm_semiring R] (s' : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) :
+  polynomial.degree (polynomial.map (eval s') f) ≤ polynomial.degree f :=
```

Comment:
Are we missing the lemma that says `(h : f.support ⊆ g.support) : f.degree ≤ g.degree`?

Reply:
Yes, it seem it was missing! I've commited this lemma and the new proof for this one

Reply:
I'd lean towards removing this proof entirely now that we have the mono lemma.

Reply:
I can remove `degree_eval_le_degree` and `nat_degree_eval_le_nat_degree` if you feel this is necessary. It seemed to me that having these may be useful.

The application I have in mind is [here](https://github.com/isadofschi/combinatorial_nullstellensatz/blob/0c0885d7ddf508c57a97826f226fef36d2800ae8/src/lemma_2_1.lean#L71). I can replace this line by
```exact polynomial.nat_degree_le_nat_degree (polynomial.degree_mono (support_eval s (fin_succ_equiv R n f))),```

Reply:
Yes, I think we should just remove them